python/ray/workflow/task_executor.py

```diff
@@ -161,3 +174,7 @@ def resolve(self, store: workflow_storage.WorkflowStorage) -> Tuple[List, Dict]:
             ray.get(a) if isinstance(a, ObjectRef) else a for a in flattened_args
         ]
         return signature.recover_args(flattened_args)
+
+    def wait_upstream_checkpoint_tasks(self):
+        """Wait until all upstream checkpoint tasks are completed successfully."""
+        ray.get(list(self.upstream_checkpoint_tasks.values()))
```

Comment:
We probably can return the size of data written so that we can know the size for each output.